docusaurus-search-local/src/server/utils/processDocInfos.ts

```diff
@@ -16,76 +18,111 @@ export function processDocInfos(
     blogRouteBasePath,
     ignoreFiles,
   }: ProcessedPluginOptions
-): DocInfoWithFilePath[] {
-  return routesPaths
-    .map<DocInfoWithRoute | undefined>((url: string) => {
-      // istanbul ignore next
-      if (!url.startsWith(baseUrl)) {
-        throw new Error(
-          `The route must start with the baseUrl "${baseUrl}", but was "${url}". This is a bug, please report it.`
-        );
+): VersionDocInfo[] {
+  const emptySet = new Set();
+  let versionData: any = [{ versionOutDir: outDir, docs: emptySet }];
+  if (plugins) {
+    const docsPluginData = plugins.find(element => element.name === "docusaurus-plugin-content-docs");
+    if (docsPluginData) {
+      versionData = [];
+      const loadedVersions:LoadedVersion[] = (docsPluginData.content as LoadedContent).loadedVersions;
+      for (const loadedVersion of loadedVersions) {
+        let docs = new Set();
+        for (const doc of loadedVersion.docs) {
+          docs.add(doc.permalink);
+        }
+        const route = loadedVersion.path.substr(baseUrl.length);
+        let versionOutDir = outDir;
+        // The last versions search-index should always be placed in the root since it is the one used from non-docs pages
+        if (!loadedVersion.isLast) {
+          versionOutDir = path.join(outDir, ...route.split("/").filter((i: string) => i));
+        }
+        versionData.push({ versionOutDir, docs });
       }
-      const route = url.substr(baseUrl.length).replace(/\/$/, "");
+    }
+  }
 
-      // Do not index homepage, error page and search page.
-      if (
-        ((!docsRouteBasePath || docsRouteBasePath[0] !== "") && route === "") ||
-        route === "404.html" ||
-        route === "search"
-      ) {
-        return;
-      }
+  // Create a list of files to index per document version. This will always include all pages and blogs.
+  let result = [];
+  for (const { versionOutDir, docs } of versionData) {
+    const versionPaths = routesPaths
```

Comment:
Yes, I thought it was fine since we just do path checking in the map function, and it made it easy to always include non-doc pages for all versions :)

Reply:
Yes, I think it would be fine, I thought we're doing file scanning there before, :)